editor/src/main/java/com/github/kitakkun/noteapp/editor/usecase/AnchorTransformer.kt

```diff
@@ -0,0 +1,167 @@
+package com.github.kitakkun.noteapp.editor.usecase
+
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.TextFieldValue
+import com.github.kitakkun.noteapp.data.model.BaseStyle
+import com.github.kitakkun.noteapp.data.model.BaseStyleAnchor
+import com.github.kitakkun.noteapp.data.model.OverrideStyle
+import com.github.kitakkun.noteapp.data.model.OverrideStyleAnchor
+import com.github.kitakkun.noteapp.editor.editmodel.EditorConfig
+import com.github.kitakkun.noteapp.editor.editmodel.TextFieldChangeEvent
+import com.github.kitakkun.noteapp.editor.ext.deleteLinesAndShiftUp
+import com.github.kitakkun.noteapp.editor.ext.insertNewAnchorsAndShiftDown
+import com.github.kitakkun.noteapp.editor.ext.optimize
+import com.github.kitakkun.noteapp.editor.ext.shiftToLeft
+import com.github.kitakkun.noteapp.editor.ext.shiftToRight
+import com.github.kitakkun.noteapp.editor.ext.splitAt
+import com.github.kitakkun.noteapp.editor.ext.toValidOrder
+
+class AnchorTransformer {
+    /**
+     * update override style anchors
+     * @param event [TextFieldValue] change event
+     * @param anchors current override style anchors
+     * @param editorConfig active editor configuration
+     */
+    fun updateOverrideAnchors(
+        event: TextFieldChangeEvent,
+        anchors: List<OverrideStyleAnchor>,
+        editorConfig: EditorConfig,
```

Comment:
This class should be tested because of its complexity. However, it seems not to be easy to do it. Refactoring this class is necessary in the future.